The logging guide review for 3.15 (#43344)
quarkusio/quarkus: docs/src/main/asciidoc/logging.adoc

:diataxis-type: reference
:topics: logging,observability

> Read about the use of logging API in {project-name}, configuring logging output, and using logging adapters to unify the output from other logging APIs.

Author: @sheilamjones Hello Sheila. Here in Quarkus, it renders as `Quarkus`, while in RHBQ repo, it renders as `Red Hat build of Quarkus`.

== Get an application logger

> In {project-name}, the most common ways to obtain an application logger are by:

Author: OK, thank you!

* <<declaring-a-logger-field>>

> * <<simplified-logging>>

Author: Originally, there was Logging with Panache, which many people find confusing; thus, we agreed to rename it that way. Besides, it is a settled term :)

<1> The FQCN of the declaring class is used as a logger name, for example, `org.jboss.logging.Logger.getLogger(SimpleBean.class)` will be used.
<2> In this case, the name _foo_ is used as a logger name, for example,`org.jboss.logging.Logger.getLogger("foo")` will be used.

NOTE: The logger instances are cached internally.
Therefore, when a logger is injected, for example, into a `@RequestScoped` bean, it is shared for all bean instances to avoid possible performance penalties associated with logger instantiation.

== Use log levels

{project-name} provides different log levels, which helps developers control the amount of information logged based on the severity of the events.

.Log levels used by {project-name}:

[horizontal]
OFF:: A special level used in configuration to turn off logging.
FATAL:: A critical service failure or total inability to handle any requests.
ERROR:: A major issue in processing or an inability to complete a request.
WARN:: A non-critical service error or problem that might not require immediate correction.

> INFO:: Service lifecycle events or other important information that occurs infrequently.

Author: I think that occurs is a frequently used term in software development, logging, and debugging. I am for keeping it but will ask a SME for an additional opinion  :)

> INFO:: Service lifecycle events or other important information that occurs infrequently.

Reviewer: I kinda agree that "information occurs" doesn't sound right. This

>  INFO:: Service lifecycle events or other important infrequent information.

sounds good to me.

> INFO:: Service lifecycle events or other important information that occurs infrequently.

Author: Great! Thank you both @Ladicek @sheilamjones . Applying! :)

----

When you set the log level to below `DEBUG`, you must also adjust the minimum log level.

> You can apply this setting either globally with the `quarkus.log.min-level` configuration property, or per category:

Reviewer 2: Are we doing second-person here? Maybe instead:
```suggestion
This setting may be applied either globally with the `quarkus.log.min-level` configuration property, or per category:
```

Reviewer 2: No, in this case keeping `may` is appropriate I think; I would use `might` for something that could happen or is allowed to happen out of the user's control (but is not certain), but use `may` for something that the user is specifically being permitted or suggested to do.
